tests/lib/prepare.sh

```diff
@@ -106,12 +106,17 @@ update_core_snap_for_classic_reexec() {
 
     # Now unpack the core, inject the new snap-exec/snapctl into it
     unsquashfs "$snap"
-    # clean the old snapd libexec binaries, just in case
-    rm squashfs-root/usr/lib/snapd/*
-    # and copy in the current ones
+    # clean the old snapd binaries, just in case
+    rm squashfs-root/usr/lib/snapd/* squashfs-root/usr/bin/{snap,snapctl}
+    # and copy in the current libexec
```

Comment:
Implementing dpkg bug after bug ;-)

Reply:
sorry I can't hear you over the noise of all my tests passing

Reply:
Is this a good time to mention tests are broken? :P

Reply:
not because of this :-D